lib_eio_linux/eio_linux.ml

```diff
@@ -1160,11 +1160,87 @@ let net = object
   method getnameinfo = Eio_unix.getnameinfo
 end
 
+let pid_to_process close pid = object
+  inherit Eio.Process.process
+  method pid = pid
+  method status = 
```

Comment:
We need to reap the child even if the caller doesn't call `status`. So status should probably just return a promise that gets resolved automatically when it finishes (this is also what Lwt does).

Reply:
`status` now returns a promise :))